plugins/org.jkiss.dbeaver.model.sql/src/org/jkiss/dbeaver/model/sql/SQLGroupingQueryGenerator.java

```diff
@@ -205,4 +211,27 @@ public String[] getFuncAliases() {
         return funcAliases;
     }
 
+    /**
+     * Represents a formatted table in an SQL query.
+     * This class is used to format the table name according to the specified SQL dialect.
+     * It is necessary because `net.sf.jsqlparser.schema.Table` only supports a dot as a separator.
+     */
+    // TODO: Remove this class when https://github.com/dbeaver/pro/issues/3140 will be resolved
+    private static class FormattedTable extends Table {
+        private final SQLDialect sqlDialect;
+
+        public FormattedTable(Table table, SQLDialect sqlDialect) {
+            super(table.getDatabase(), table.getSchemaName(), table.getName());
+            this.sqlDialect = sqlDialect;
+        }
+
+        @Override
+        public String toString() {
+            String databaseName = getDatabase().getDatabaseName() != null ? getDatabase().getDatabaseName() + sqlDialect.getCatalogSeparator() : "";
```

Comment:
Potentially it is possible that database name will be empty (""). Use CommonUtils.isEmpty